### PR TITLE
#5 Support branch name input

### PR DESCRIPTION
**Pull request by salesh**
#5 
Hi Manuel 

1) 'B','b', and 'm' was taken - and I was thinking like 'M' should be fine and closest
2) Should default be main or master? :/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,6 +89,8 @@ OPTIONS
 
   -b, --branch=branch                  [default: wax] the branch name where apply the changes
 
+  -M, --main-branch=main-branch        [default: main] the main branch in the origin repo
```
Comment (Eomm) [inline README.md:92]: I would keep `master` as default just to release this PR as semver minor

```suggestion
  -O, --pr-origin=master        [default: master] the target PR branch name in the origin repository
```

File: README.md [hunk 2]
```diff
@@ -111,13 +113,13 @@ DESCRIPTION
   - clone the repos in your local env
   - process all the files of the cloned repos
   - commit the changes in a dedicated branch
-  - open a PR to the master branch in the origin repo
+  - open a PR to the main branch in the origin repo
```
Comment (Eomm) [inline README.md:116]:
```suggestion
  - open a PR to the `pr-origin` branch in the origin repo
```

File: README.md [hunk 3]
```diff
@@ -111,13 +113,13 @@ DESCRIPTION
...
 
   All these steps are optionals.
 
 EXAMPLES
   Change all the LICENCE file of your org
-    $ upgrade -K GITHUB-TOKEN --fork -p='./toMit.js' -t='Changed license' -c='chore changed license' -b licensebranch -r 
+    $ upgrade -K GITHUB-TOKEN --fork -p='./toMit.js' -t='Changed license' -c='chore changed license' -M main -b licensebranch -r
```
Comment (Eomm) [inline README.md:122]:
```suggestion
    $ upgrade -K GITHUB-TOKEN --fork -p='./toMit.js' -t='Changed license' -c='chore changed license' -O main -b licensebranch -r 
```

File: src/commands/upgrade.js [hunk 4]
```diff
@@ -119,6 +119,11 @@ UpgradeCommand.flags = {
     description: 'the branch name where apply the changes',
     default: 'wax'
   }),
+  main_branch: flags.string({
+    char: 'M',
+    description: 'the main branch in the origin repo',
+    default: 'main'
```
Comment (Eomm) [inline src/commands/upgrade.js:125]:
```suggestion
    default: 'master'
```

File: src/commands/upgrade.js [hunk 5]
```diff
@@ -119,6 +119,11 @@ UpgradeCommand.flags = {
     description: 'the branch name where apply the changes',
     default: 'wax'
   }),
+  main_branch: flags.string({
```
Comment (Eomm) [inline src/commands/upgrade.js:122]: sneak case 😝 

```suggestion
  'pr-origin': flags.string({
```

File: src/commands/upgrade.js [hunk 6]
```diff
@@ -162,7 +167,7 @@ UpgradeCommand.flags = {
 
 UpgradeCommand.examples = [
   'Change all the LICENCE file of your org',
-  " $ upgrade -K GITHUB-TOKEN --fork -p='./toMit.js' -t='Changed license' -c='chore changed license' -b licensebranch -r repo-list.txt"
+  " $ upgrade -K GITHUB-TOKEN --fork -p='./toMit.js' -t='Changed license' -c='chore changed license' -M main -b licensebranch -r repo-list.txt"
```
Comment (Eomm) [inline src/commands/upgrade.js:170]:
```suggestion
  " $ upgrade -K GITHUB-TOKEN --fork -p='./toMit.js' -t='Changed license' -c='chore changed license' -O main -b licensebranch -r repo-list.txt"
```

File: test/upgrade.test.js [hunk 7]
```diff
@@ -104,6 +104,7 @@ test('Fork Clone Commit PR', t => {
     '-r', 'https://github.com/pkgjs/support',
     '-m', 'json$',
     '-b', 'branch-name',
+    '-M', 'main-branch-name',
```
Comment (Eomm) [inline test/upgrade.test.js:107]:
```suggestion
    '-O', 'main-branch-name',
```

File: src/commands/upgrade.js [hunk 8]
```diff
@@ -64,7 +64,7 @@ class UpgradeCommand extends Command {
       if (flags.pr) {
         const prCoordinates = {
           head: `${fork.owner || repo.owner}:${flags.branch}`,
-          base: 'master' // TODO destination repo
+          base: flags.main_branch
```
Comment (Eomm) [inline src/commands/upgrade.js:67]:
```suggestion
          base: flags['pr-origin']
```

**Review by Eomm — COMMENTED** [review]
Thanks for submitting this PR!

To answer:

- I would rename the arg in order to step away from branch naming: the parameter is needed to target the right remote branch
- To keep this PR as semver-minor, `master` should be the default. We can change it in the next major version I would like to bump within a complete project refresh

File: src/commands/upgrade.js (hunk 5 above)
Comment (salesh) [reply src/commands/upgrade.js]: Sorry for that one :/

**Comment by salesh** [comment]
> Thanks for submitting this PR!
> 
> To answer:
> 
> * I would rename the arg in order to step away from branch naming: the parameter is needed to target the right remote branch
> * To keep this PR as semver-minor, `master` should be the default. We can change it in the next major version I would like to bump within a complete project refresh

Thank you very much for clarifying and explaining to me everything

**Review by Eomm — APPROVED** [review]

**Pull request merged**